### ISSUE TITLE
Oppkjøp av DnB: Alternativer med god API tilgang til data?

### ISSUE THREAD

**Issue by yonyonson** [stenjo/SbankenToYNAB#35]
Hei,

vet ikke om dette er innenfor omfanget at denne repo, men SBanken skal kjøpes opp av DnB, og såklart kommer DnB å beholde alle grensesnitter med det første, men det er viss tvil om de er så dedikert til å beholde de på lengre sikt.
Jeg vurderer derfor å bytte bank, men vil gjerne beholde integrasjon med YNAB slik den er i dag.

Spørsmål er: Har noen sett etter bank med tilsvarende god API tilgang til konto- og transaksjonsinformasjon? 

Lukk ticket om den er ikke relevant, @stenjo .

**Comment by steoj** [comment]
Svært relevant, og har tenkt på dette selv. Oppkjøpet skaper litt usikkerhet rundt hvordan dette API'et videreutvikles og vedlikeholdes.
Det er en prosess på europeisk nivå som krever åpne api-er til banker men usikker på hvor langt det er kommet: https://www.bloomberg.com/professional/blog/europes-new-api-rules-lay-groundwork-for-regulating-open-banking/
Tror det er referert til som PSD2 eller Betalingstjenestedirektivet her til lands og jeg ser flere banker tilbyr åpne api'er nå

**Comment by yonyonson** [comment]
Ja. Jeg har forsøkt såvidt å se på PSD2 grensesnitter, og det er AISP (Account Information Service Provider) som er interessant. Den er blitt norsk lov i 2020, og alle banker er pålagt å implementere PSD2 innen april 2021.

Samtidig virker som om PSD2 deler ikke mellom rolle som kreves for å bruke grensesnittet for betalingstjenestetilbyder (TISP) og kontoinformasjonstilbyder (AISP), og begge krever at organisasjon som bruker den er en tiltrodde tredjepart (TTP). Det virker som om Finanstilsynet må godkjenne alle TTPer, og det å være en TTP er en forutsetning for at man skal kunne ta i bruk PSD2 grensesnitter.
Om jeg tolker dette riktig (og det kan godt hende at jeg tar feil her), men dette betyr for meg at PSD2 gresesnitter er ikke tiltenkt for at enklebrukerne som oss skal kunne ta de i bruk for å hente sine data (noe som er veldig skuffende).

Og her er en eksmepel på sertifikat som kreves for å være TTP, eIDAS QTSP fra BuyPass: https://www.buypass.no/produkter/virksomhetssertifikat-esegl/psd2-rts-godkjente-sertifikater-for-sterk-autentisering-qc-eseal

Jeg er foreløpig noe forvirret.

## Kilder

* Årsrapport for Finanstilsynet for 2021 (søk etter ordet TTP): https://www.finanstilsynet.no/publikasjoner-og-analyser/arsrapport/arsrapport-2021/rapporter-fra-tilsynsomradene-for-2021/banker-og-annen-finansieringsvirksomhet/
* Det oganisatoriske / jurisike fra Finanstilsynet: https://www.finanstilsynet.no/tema/psd-2---eus-reviderte-betalingstjenestedirektiv/
* Oversikt over utviklersportaler for forskjellige banker hos BITS: https://www.bits.no/bank/developer-portals-psd2/

**Comment by yonyonson** [comment]
Og her er en eksmepel på sertifikat som kreves for å være TTP, eIDAS QTSP fra BuyPass: https://www.buypass.no/produkter/virksomhetssertifikat-esegl/psd2-rts-godkjente-sertifikater-for-sterk-autentisering-qc-eseal

**Comment by yonyonson** [comment]
Alternativ veien å se er Open Banking Intermediaries, hvor Forrester skrev en rapport om dette i Q12021: https://www.readkong.com/page/the-forrester-new-wave-open-banking-intermediaries-q1-2021-6247081

Og da gir dette mening å se på [Plaid](https://plaid.com/en-eu/), [Tink](https://tink.com/) (svensk), [Bud](https://www.thisisbud.com/), og [TrueLayer](https://truelayer.com/). Alle de krever ekstra betaling for å gi tilgang til info, men dette kan også være en løsning.

Jeg ville ha sett litt på hvordan de verner personopplysninger etter GDPR eller tilsvarende, og ville ha valgt blant disse, muligens. Da kommer dette å definere en forutsetning at den som tar denne integrasjonen i bruk kjøper tilgang til API fra en av disse leverandører, men da kan data hentes fra flere banker, ikke kun fra SBanken.

*Oppdatering*: Det finnes påbegynte integrasjoner fra Tink: https://github.com/fmfdias/tink-ynab og TrueLayer: https://github.com/LunNova/import-ynab og Plaid: https://github.com/colmdoyle/plaid-ynab , men ingen av de ser noe særlig modne ut.
